src/streamlink/session.py

```diff
@@ -257,6 +257,9 @@ def set_option(self, key, value):
 
         if key == "http-proxy":
             self.http.proxies["http"] = update_scheme("http://", value)
+            if "https" not in self.http.proxies:
+                self.http.proxies["https"] = update_scheme("http://", value)
```

Comment:
not sure if it changes anything,
but `update_scheme("http://", value)` is `https` two lines below instead of `http`

Reply:
It’ll set the default scheme for the url, for this I figured they should both default to `http`, it would be unusual to have to different schemes for the same proxy I think...